genum/gen/traits.go

```diff
@@ -15,11 +32,123 @@ func (s TraitDescs) Less(i, j int) bool {
 	return s[i].Name < s[j].Name
 }
 
+func (s TraitDescs) getParsableUnderlying(u Underlying) TraitDescs {
+	out := make([]TraitDesc, 0, len(s))
+	for _, t := range s {
+		if t.Parsable && t.hasUnderlying(u) {
+			out = append(out, t)
+		}
+	}
+	return out
+}
+
+func (s TraitDescs) GetParsableUnderlyingString() TraitDescs {
+	return s.getParsableUnderlying(String)
+}
+
+func (s TraitDescs) GetParsableUnderlyingFloat64() TraitDescs {
+	return s.getParsableUnderlying(Float64)
+}
+
+func (s TraitDescs) GetParsableUnderlyingFloat32() TraitDescs {
+	return s.getParsableUnderlying(Float32)
+}
+
+func (s TraitDescs) GetParsableUnderlyingInt64() TraitDescs {
+	return s.getParsableUnderlying(Int64)
+}
+
+func (s TraitDescs) GetParsableUnderlyingUint64() TraitDescs {
+	return s.getParsableUnderlying(String)
+}
```

Comment:
sadly, these need to ensure the underlying type does not implement their respective yaml/json/text types when in those funcs... larger write up somewhere else in this pr.